ios/RNCameraRoll.m

```diff
@@ -60,86 +85,96 @@ @implementation RNCameraRoll
         } else if (adjDate == nil && [epochNSDate timeIntervalSinceDate:creDate] > 0) {
           // if creation timestamp is less than our filter and the image has never been modified, return
           return;
-        } else if ( [extension caseInsensitiveCompare:@"heic"] == NSOrderedSame ) {
-          [self _processHEIC:asset orgFilename:orgFilename];
         } else {
-          [self _processJPG:asset orgFilename:orgFilename];
+          dispatch_group_enter(assetDatasGroup);
+          [self _writeToDisk:asset orgFilename:orgFilename onComplete:^(NSString *path, UIImageOrientation orientation) {
+            if (path) {
+              AssetData *data = [[AssetData alloc] init];
+              data.asset = asset;
+              data.path = path;
+              data.orientation = orientation;
+              [assetDatas addObject:data];
+            }
+            dispatch_group_leave(assetDatasGroup);
+          }];
         }
       }];
+
+      // assetDatas fully populated now
+      [assetDatas sortUsingSelector:@selector(compare:)];
+
+      NSMutableArray<NSDictionary*> *results = [[NSMutableArray alloc] initWithCapacity:assetDatas.count];
+
+      [assetDatas enumerateObjectsUsingBlock:^(AssetData *assetData, NSUInteger idx, BOOL *stop) {
+        [results addObject:[self _convertAssetData:assetData]];
+      }];
+
+      NSError *serializationError;
+      NSData *data = [NSJSONSerialization dataWithJSONObject:results options:NSJSONWritingPrettyPrinted error:&serializationError];
+      if(serializationError) {
+        reject(@"0", serializationError.localizedFailureReason, serializationError);
+      } else {
+        NSString* jsonStr = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+        resolve(jsonStr);
```

Comment:
Convert to a jsonString and resolve the promise... Do we need to convert to jsonString or can we just resolve with an array of NSDictionary? Will have to look that up.